lib/libtlspool.c

```diff
@@ -37,6 +38,18 @@ static pthread_cond_t updated_poolfd = PTHREAD_COND_INITIALIZER;
 
 static pthread_mutex_t prng_lock = PTHREAD_MUTEX_INITIALIZER;
 
+int tlspool_getpid () {
+	int fd;
+
+	if ((fd = open("/var/run/tlspool.pid", O_RDONLY)) != -1) {
```

Comment:
Hardcoded PID-filenaam is strijdig met configureerbaarheid ervan in tlspool.conf.
Het is me niet helemaal duidelijk waarom je niet getpid() gebruikt?
Dat is tenminste waar het vandaan komt, en ik kan me geen geval voorstellen waarin een andere uitkomst zou komen op het moment dat je getpid() later aanroept.

Als je toch een gesavede versie van getpid() ten tijde van schrijven van deze file nodig hebt, dan stel ik voor die in config.c naar een static global te schrijven, en deze function op te nemen in config.c als

static global int tlspool_pid;

void cfg_pidfile (...) {
    ...
    tlspool_pid = getpid ();
   snprintf (pidbuf, sizeof (pidbuf)-1, "%d\n", tlspool_pid);
   ...
}

int cfg_filed_pid (void) {
   return tlspool_pid;
}